systemverilog-plugin/UhdmAst.cc

```diff
@@ -3270,7 +3270,12 @@ void UhdmAst::process_logic_var()
         }
         delete node;
     });
-    visit_one_to_many({vpiRange}, obj_h, [&](AST::AstNode *node) { packed_ranges.push_back(node); });
+    if (auto typespec_h = vpi_handle(vpiTypespec, obj_h)) {
+        visit_one_to_many({vpiRange}, typespec_h, [&](AST::AstNode *node) { packed_ranges.push_back(node); });
+        vpi_release_handle(typespec_h);
+    } else {
+        visit_one_to_many({vpiRange}, obj_h, [&](AST::AstNode *node) { packed_ranges.push_back(node); });
+    }
```

Comment:
Generally LGTM, but I think this pattern is common now, so maybe it is worth to extract it to some function?

Reply:
Looking through code, sometimes typespec handle is obtained earlier (and checked for type), sometimes we want a fallback, sometimes don't. This exact flow is present just once, so I don't think it should be extracted now, but let's definitely keep that in mind for the future (added a `TODO`).

Reply:
OK, fine by me.